bioptim/interfaces/acados_interface.py

```diff
@@ -105,30 +107,49 @@ def __set_constrs(self, ocp):
         ## TODO: implement constraints in g
 
         # path control constraints
+        self.x_bound_max = []
+        self.x_bound_min = []
+        d = [-1, 0, 1]
+        if ocp.nlp[0].parameters_to_optimize:
+            for key in ocp.nlp[0].parameters_to_optimize.keys():
+                for i in range(3):
+                    self.x_bound_max.append(np.concatenate(
+                        (np.concatenate([ocp.nlp[0].parameters_to_optimize[key]['bounds'].max[0]] * ocp.nlp[0].np),
+                         np.array(ocp.nlp[0].x_bounds.max[:, d[i]]))))
+
```

Comment:
does not work if several params: it would concatenate several times ocp.nlp[0].x_bounds.max[:, d[i]].
Consider removing this concatenate and...